examples/README.md

```diff
@@ -0,0 +1,179 @@
+you will be managing it through the
+[Astarte Dashboard](https://github.com/astarte-platform/astarte-dashboard).
+The `toggle_led` example will assume the use of a local instance of Astarte.
+Specifically, it assumes you created one following the
+[Astarte in 5 minutes](https://docs.astarte-platform.org/latest/010-astarte_in_5_minutes.html) guide.
+In *Astarte in 5 minutes* you can stop right before installing the interfaces step.
+
+# Common configuration
+
+All the examples have some common configuration required.
+
+## Installing the interfaces on Astarte (for the remote instance)
+
+The interfaces used by each example shall be first installed on the remote Astarte instance.
+
+To install a new interface using the Astarte Dashboard follow this sequence of steps:
+1. In the *Interfaces* tab click on *Install a new interface ...*
+2. In the interface editor fill in the required information for your interface. If you already have
+an interface definition in the `.json` format you can paste that directly into the editor on the
+right of the page.
+3. Finish by clicking on install interface
+4. Refresh the page to check that the interface has been installed correctly.
+
+Each example contains an `/interfaces` folder. To run that example install all the interfaces
+contained in the `.json` files in such folder.
+
+## Installing the interfaces on Astarte (for the local instance)
+
+We shall install the required interfaces in our local Astarte instance.
+
+An interface can be installed by running the following command:
+
+```
+astartectl realm-management                       \
+    --realm-management-url http://localhost:4000/ \
+    --realm-key <REALM>_private.pem               \
+    --realm-name <REALM>                          \
+    interfaces install <INTERFACE_FILE_PATH>
+```
+Where `<REALM>` is the name of the realm, and `<INTERFACE_FILE_PATH>` is the path name to the
+`.json` file containing the interface description.
+
+Each example contains an `/interfaces` folder. To run that example install all the interfaces
+contained in the `.json` files in such folder.
+
+## Registering a new device on Astarte (only when manually registering a device)
+
+For our device to connect correctly to Astarte it shall be registered in our realm. The registration
+ will produce a credential secret.
+
+To manually register the device on Astarte Cloud you can use the dashboard, following this
+procedure:
+1. In the *Devices* tab click on *Register a new device*
+2. Click *Generate random ID* and then on *Register device*
+3. Copy the device credential secret somewhere local
+
+**NB**: While the device ID can always be recovered from the Astarte Dashboard, the
+credential secret is only shown once during the device registration procedure.
+
+## Generating the Pairing JWT (only when auto registering a device)
+
+We will now generate a Parining JWT from our Astarte local instance. This JWT will be used
+by the device during the first device registration.
```

Comment:
Is it possible to register a device twice? Otherwise, I'd go with "*the initial device registration*"